src/Raven.Server/ServerWide/ShardingStore.cs

```diff
@@ -125,9 +134,37 @@ public string ReadPublishedUrls()
                 HttpPooledConnectionLifetime = DocumentConventions.DefaultForServer.HttpPooledConnectionLifetime,
                 UseHttpCompression = _serverStore.Configuration.Sharding.ShardExecutorUseHttpCompression,
                 UseHttpDecompression = _serverStore.Configuration.Sharding.ShardExecutorUseHttpDecompression,
-                GlobalHttpClientTimeout = _serverStore.Configuration.Sharding.OrchestratorTimeoutInMinutes.AsTimeSpan
+                GlobalHttpClientTimeout = _serverStore.Configuration.Sharding.OrchestratorTimeoutInMinutes.AsTimeSpan,
+                CreateHttpClient = handler =>
+                {
+                    handler.ServerCertificateCustomValidationCallback = ShardingCustomValidationCallback;
+                    return new HttpClient(handler);
+                }
             };
 
+        private bool ShardingCustomValidationCallback(HttpRequestMessage message, X509Certificate2 cert, X509Chain chain, SslPolicyErrors errors)
+        {
+            if (cert.Thumbprint == _serverStore.Server.Certificate.Certificate.Thumbprint)
+                return true;
+
+            var pinningHash = cert.GetPublicKeyPinningHash();
+
+            using (_serverStore.Engine.ContextPool.AllocateOperationContext(out ClusterOperationContext context))
+            using (context.OpenReadTransaction())
+            {
+                foreach (var certificateDefinition in _serverStore.Cluster.GetCertificatesByPinningHashSortedByExpiration(context, pinningHash))
+                {
+                    using (var goodKnownCert = CertificateLoaderUtil.CreateCertificate(Convert.FromBase64String(certificateDefinition.Certificate)))
+                    {
+                        if (CertificateUtils.CertHasKnownIssuer(cert, goodKnownCert, _serverStore.Configuration.Security))
+                            return true;
+                    }
+                }
+            }
+
```

Comment:
This is wrong - you need to compare the `cert.GetPublicKeyPinningHash()` to `_serverStore.Server.Certificate.Certificate.GetPublicKeyPinningHash()`

Reply:
fixed